src/operator/core/crdHandler.go

```diff
@@ -71,39 +72,43 @@ func (dm *KVMSOperator) UnMapLabelIdentity(identity uint16, ewName, label string
 		log.Fatal(err)
 	}
 
-	identities := dm.MapLabelToIdentities[label]
-	for index, value := range dm.MapLabelToIdentities[label] {
-		if value == identity {
-			identities[index] = identities[len(identities)-1]
-			identities[len(identities)-1] = 0
-			identities = identities[:len(identities)-1]
+	for _, label := range labels {
+		identities := dm.MapLabelToIdentities[label]
+		for index, value := range dm.MapLabelToIdentities[label] {
+			if value == identity {
+				identities[index] = identities[len(identities)-1]
+				identities[len(identities)-1] = 0
+				identities = identities[:len(identities)-1]
+			}
+		}
+		// After deleting the identity from the label map
+		// update the etcd with updated label to identities map
+		mapStr, _ := json.Marshal(dm.MapLabelToIdentities[label])
+		err = dm.EtcdClient.EtcdPut(context.TODO(), ct.KvmOprLabelToIdentities+label, string(mapStr))
+		if err != nil {
+			log.Fatal(err)
 		}
-	}
-	// After deleting the identity from the label map
-	// update the etcd with updated label to identities map
-	mapStr, _ := json.Marshal(dm.MapLabelToIdentities[label])
-	err = dm.EtcdClient.EtcdPut(context.TODO(), ct.KvmOprLabelToIdentities+label, string(mapStr))
-	if err != nil {
-		log.Fatal(err)
 	}
 }
 
-func (dm *KVMSOperator) UpdateIdentityLabelsMap(identity uint16, label string) {
-	kg.Printf("Updating identity to label map identity:%d label:%s", identity, label)
-	dm.MapIdentityToLabel[identity] = label
-	err := dm.EtcdClient.EtcdPut(context.TODO(), ct.KvmOprIdentityToLabel+strconv.FormatUint(uint64(identity), 10), label)
-	if err != nil {
-		log.Fatal(err)
-	}
-
-	_, found := Find(dm.MapLabelToIdentities[label], identity)
-	if !found {
-		dm.MapLabelToIdentities[label] = append(dm.MapLabelToIdentities[label], identity)
-		mapStr, _ := json.Marshal(dm.MapLabelToIdentities[label])
-		err := dm.EtcdClient.EtcdPut(context.TODO(), ct.KvmOprLabelToIdentities+label, string(mapStr))
+func (dm *KVMSOperator) UpdateIdentityLabelsMap(identity uint16, labels []string) {
+	for _, label := range labels {
+		kg.Printf("Updating identity to label map identity:%d label:%s", identity, label)
+		dm.MapIdentityToLabel[identity] = label
+		err := dm.EtcdClient.EtcdPut(context.TODO(), ct.KvmOprIdentityToLabel+strconv.FormatUint(uint64(identity), 10), label)
 		if err != nil {
 			log.Fatal(err)
 		}
+
+		_, found := Find(dm.MapLabelToIdentities[label], identity)
+		if !found {
+			dm.MapLabelToIdentities[label] = append(dm.MapLabelToIdentities[label], identity)
+			mapStr, _ := json.Marshal(dm.MapLabelToIdentities[label])
+			err := dm.EtcdClient.EtcdPut(context.TODO(), ct.KvmOprLabelToIdentities+label, string(mapStr))
+			if err != nil {
+				log.Fatal(err)
```

Comment:
Using Fatal may not be a good idea in general. We should look into all the Fatals and use errors and exit gracefully.